android/src/main/java/com/ajinasokan/flutterdisplaymode/DisplayModePlugin.java

```diff
@@ -0,0 +1,180 @@
+                setDefaultMode(result);
+                break;
+            default:
+                result.notImplemented();
+                break;
+        }
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    private Display.Mode getCurrentMode() {
+        final WindowManager windowManager = (WindowManager) activity.getSystemService(Context.WINDOW_SERVICE);
+        final Display display = windowManager.getDefaultDisplay();
+        return display.getMode();
+    }
+
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    private void getCurrentMode(@NonNull Result result) {
+        final Display.Mode mode = getCurrentMode();
+        final HashMap<String, Object> ret = new HashMap<>();
+        ret.put("id", mode.getModeId());
+        ret.put("width", mode.getPhysicalWidth());
+        ret.put("height", mode.getPhysicalHeight());
+        ret.put("refreshRate", mode.getRefreshRate());
+        ret.put("selected", true);
+        result.success(ret);
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    private ArrayList<HashMap<String, Object>> getSupportedModes() {
+        final ArrayList<HashMap<String, Object>> ret = new ArrayList<>();
+        final WindowManager windowManager = (WindowManager) activity.getSystemService(Context.WINDOW_SERVICE);
+        final Display display = windowManager.getDefaultDisplay();
+        final Display.Mode[] modes = display.getSupportedModes();
+        if (modes == null) {
+            return ret;
+        }
+
+        final Window window = activity.getWindow();
+        final WindowManager.LayoutParams params = window.getAttributes();
+        final int selectedMode = params.preferredDisplayModeId;
+
+        for (final Display.Mode mode : modes) {
+            final HashMap<String, Object> item = new HashMap<>();
+            item.put("id", mode.getModeId());
+            item.put("width", mode.getPhysicalWidth());
+            item.put("height", mode.getPhysicalHeight());
+            item.put("refreshRate", mode.getRefreshRate());
+            item.put("selected", mode.getModeId() == selectedMode);
+            ret.add(item);
+        }
+        return ret;
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    private void getSupportedModes(@NonNull Result result) {
+        result.success(getSupportedModes());
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    private void setMode(@NonNull MethodCall call, @NonNull Result result) {
+        if (call.argument("mode") != null) {
+            final int mode = (int) Objects.requireNonNull(call.argument("mode"));
+            final Window window = activity.getWindow();
+            final WindowManager.LayoutParams params = window.getAttributes();
+            params.preferredDisplayModeId = mode;
+            window.setAttributes(params);
+            result.success(null);
+        }
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    private void setDefaultMode(@NonNull Result result) {
```

Comment:
I'm confused. How is this different from `setMode` function? How does this set the mode as default?

Reply:
It only worked before a mode was set manually, read the current mode using `getCurrentMode` from system, then set it as current mode (because no mode was `selected`). After any mode was set this will take no effect.

Reply:
I misunderstood. I thought it was to set a default mode for the app so that it is persisted across sessions. Your fix can be merged into the getSupportedModes function I guess. We don't have to re-apply the current mode to updated preferred mode id.

Reply:
> We don't have to re-apply the current mode to update preferred mode id.

I got your point, but it's a little different that cannot be merged into `getSupportedModes` as I thought. What if the user didn't called `getSupportedModes`? We can make an assertion or failed result if there's a mode that has been selected.

Reply:
I think we don't have to worry about that. Dev has to call getSupportedModes to set one mode. That's why I kept the param for setMode to the Mode class instead of an int id.

Reply:
Let's do this:
* Remove `setDefaultMode`
* Remove `selected` field from Mode
So we will have three functions get all modes, get currently selected, and to select new one. I think that is cleaner. What do you think?

Reply:
Hmm...so the user needs to call `FlutterDisplayMode.setMode(FlutterDisplayMode.current)` to set a default mode? That seems too  coupled and complex that needs two step to achieve this.
My first thought is:
```dart
assert(supported.where((m) => m.selected, orElse: () => null).isEmpty, 'You already set a default mode.')
```

Reply:
Again I'm confused. Why a call to `FlutterDisplayMode.setMode(FlutterDisplayMode.current)` is required?

Reply:
In a scenario like me, using OnePlus, I cannot run my flutter app in the same display mode if the refresh rate of my phone is not 60Hz. That requires me to run set mode manually every time the app start, to make sure it runs properly.

Reply:
Oh my bad. I keep forgetting this. I think the part that confuses me is the name of that function. What about something like "setDisplayDefault" or "setDeviceDefault"?

Reply:
setDeviceDefault looks good to me.